### ISSUE TITLE
Add a warning for unresolved constructor name

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#7660]
```agda
open import Agda.Builtin.Bool
open import Agda.Builtin.Nat

data D1 : Set where
  c : Bool → D1

data D2 : Set where
  c : Nat → D2

test : _ → Set
test (c x) = {!   !}

test2 : _
test2 = c {!   !}
```
I was rather surprised that Agda does not report an error but only reports some rather cryptic unsolved metavariables:
```
*All Goals, Errors*
Sort _6
_7 : _6
Sort _8
_9 : _8
_11 : _9
```
Also both holes report "No type nor action available for hole ?0." when trying to interact with them.

It would be nice if we could give some kind of indication (an error warning?) that the problem is due to the unresolved overloading of the name `c`.

**Comment by andreasabel** [comment]
For the lhs case, `test`, there is some more informative error.

For the rhs case, `test2`, the information is suppressed since 2.6.1.  Someone decided we should no longer print postponed type-checking problems.
Agda 2.6.0 still had:
```
Failed to solve the following constraints:
  _8 := c ? :? _7
```

**Comment by plt-amy** [comment]
Was someone me? It feels like it might've been me. I still think that's true, by the way: I think we should turn these constraints into actual warnings (could be right at the end, when we produce the list of warnings) so users have something searchable to go off of.

**Comment by jespercockx** [comment]
> For the lhs case, `test`, there is some more informative error.

What more informative error are you getting? I seem to just get unsolved metas and nothing else.

> Was someone me?

I guess the problem is the same as in https://github.com/agda/agda/issues/4519, which would mean it was probably introduced by @Saizan 's work on https://github.com/agda/agda/issues/3606.

**Comment by andreasabel** [comment]
> What more informative error are you getting? I seem to just get unsolved metas and nothing else.

I am seeing:
```
error: [UnsolvedConstraints]
Failed to solve the following constraints:
  Check definition of test : _7 → Set
    stuck because
      LOCATION: error: [SplitError.BlockedType]
      Cannot split on argument of unresolved type _7
      when checking that the pattern c x has type _7
    (blocked on _7)
```
where `LOCATION` points to the pattern `(c x)`.

**Comment by jespercockx** [comment]
I tried again and indeed the first function does give the error you mention, but the second one doesn't. In both cases it would still be nice to have a hint that the problem is due to an overloaded constructor and could be solved by qualifying it.
